torch/csrc/jit/codegen/cuda/scheduler/registry.cpp

```diff
@@ -266,6 +267,68 @@ class SchedulerTopologyChecker {
     }
     return false;
   }
+
+  // Checks if there's any unsupported operations post reduction. If outer
+  // reduction we can fuse some pointwise ops if they don't require
+  // broadcasting. For inner reductions we cannot fuse any binary like
+  // operations. Included other operations we're not even fusing at the moment
+  // like transpose and shift.
```

Comment:
Outer reduction fusions post reduction are generally perf safe(r). I left the if inner, in case we do need some separate else logic though which is probably why it seems like it's missing something.

Reply:
Ah just need to remove outer reduction from the comment, will update it.